packages/model-viewer/src/features/scene-graph/pbr-metallic-roughness.ts

```diff
@@ -56,44 +56,47 @@ export class PBRMetallicRoughness extends ThreeDOMElement implements
       pbrMetallicRoughness.metallicFactor = 1;
     }
 
-    let {baseColorTexture, metallicRoughnessTexture} = pbrMetallicRoughness;
-    const baseColorTextures = new Set<ThreeTexture>();
-    const metallicRoughnessTextures = new Set<ThreeTexture>();
+    const baseColorTexture =
+        pbrMetallicRoughness.baseColorTexture ?? {index: -1};
+
+    const metallicRoughnessTexture =
+        pbrMetallicRoughness.metallicRoughnessTexture ?? {index: -1};
+
+    const baseColorTextures: ThreeTexture|null =
+        correlatedMaterials.values().next().value ?
+        correlatedMaterials.values().next().value.map :
+        null;
+
+    // NOTE: GLTFLoader users the same texture for metalnessMap and
+    // roughnessMap in this case
+    // @see https://github.com/mrdoob/three.js/blob/b4473c25816df4a09405c7d887d5c418ef47ee76/examples/js/loaders/GLTFLoader.js#L2173-L2174
+    const metallicRoughnessTextures: ThreeTexture|null =
+        correlatedMaterials.values().next().value ?
+        correlatedMaterials.values().next().value.metalnessMap :
```

Comment:
For consistency, might be nice to use the same destructing method you used above for the textures on material.

Reply:
Sorry, I'm not sure what you mean by destructing method.

Reply:
With emissiveTexture, normalTexture, etc, you pulled them all out at once. Pretty sure this code does exactly the same thing, it just looks different. It's a tad easier to read when the same operation looks the same. Just a nit.

Reply:
Ah I see